include/fluent-bit/flb_output.h

```diff
@@ -428,6 +432,28 @@ struct flb_output_flush {
     struct mk_list _head;              /* Link to flb_task->threads */
 };
 
+/*
+ * stores timer coros on the timer_coro_list, if the output uses them
+ */
+struct flb_output_timer_coro {
```

Comment:
Ok. I think output is still needed in the name since in the future we could have timer coros for other plugin types as well I think.

So I think I'd do `flb_async_timer.c` and `flb_out_async_timer...`